### ISSUE TITLE
phe_life_expectancy: deaths and populations field to include summative totals

### ISSUE THREAD

**Issue by ianbatesncc** [PublicHealthEngland/PHEindicatormethods#33]
First many thanks for a package I find useful.

**Feature request/enhancement**

### In Brief

I would find it useful if the return value from the `phe_life_expectancy` function included summative count totals, rather than ageband specific totals, in a similar way that the return value from the `phe_dsr` function includes `total_count` and `total_pop` fields.

### Detail

I am suggesting that the `{{population}}` and `{{deaths}}` fields in the LE return value contain cumulative sum totals relating to the ageband LE calculation rather than just the ageband specific counts.

The following exemplifies my suggestion using data from the example in `phe_life_expectancy` documentation.

```
library("dplyr")
library("PHEindicatormethods")

# A simple example
df <- data.frame(startage = c(0L, 1L, 5L, 10L, 15L, 20L, 25L, 30L, 35L, 40L, 45L, 50L, 55L,
                              60L, 65L, 70L, 75L, 80L, 85L, 90L),
                 pops = c(7060L, 35059L, 46974L, 48489L, 43219L, 38561L, 46009L, 57208L,
                          61435L, 55601L, 50209L, 56416L, 46411L, 39820L, 37978L,
                          37039L, 33288L, 23306L, 11936L, 11936L),
                 deaths = c(17L, 9L, 4L, 8L, 20L, 15L, 24L, 33L, 50L, 71L, 100L, 163L,
                            263L, 304L, 536L, 872L, 1390L, 1605L, 1936L, 1937L))
```

#### Current output
Here I am calculating LE at birth and at age 65.

The LE values are calculated over the whole input dataset however the pops and deaths data shown relate only to the specific ageband(s) returned. 
```
le_b65 <- phe_life_expectancy(df, deaths, pops, startage, le_age = c(0, 65), type = "standard")
le_b65

##   startage  pops deaths    value  lowercl  uppercl
## 1        0  7060     17 80.16961 79.88269 80.45653
## 2       65 37978    536 18.76857 18.57571 18.96143
```

#### Cumulative events and denominator
Here I'm doing the same LE calculation however I've created cumulative count totals in the input data.frame before calculating the LE values so they show through in the return value:
```
df2 <- df %>%
    arrange(desc(startage)) %>%
    mutate_at(vars(pops, deaths), c(cmltv = cumsum)) %>%
    arrange(startage)

le2_b65 <- phe_life_expectancy(
    df2, deaths, pops, startage, le_age = c(0, 65), type = "standard"
)
le2_b65

##   startage  pops deaths pops_cmltv deaths_cmltv    value  lowercl  uppercl
## 1        0  7060     17     787954         9357 80.16961 79.88269 80.45653
## 2       65 37978    536     155483         8276 18.76857 18.57571 18.96143
```
I find it more relevant and useful to know the information in the `_cmltv` fields, e.g. that the 'at birth' LE value involved 9,300 events in a population of 780,000, rather than the ageband specific information that there were 17 events in a population of 7,000 aged 0 (which is readily available in the input dataset).

#### Discussion

There is the workaround to 'do-it-myself', i.e. creating the cumulative totals in the input dataset in the first place, which is OK (and is what I am doing now).  

I argue that cumulative totals are more relevant than ageband specific counts in terms of the LE return value.  In much the same way that that with the `phe_dsr` return value the `total_count` and `total_pop` fields are relevant.  

I leave this idea here for you to consider if you wish to take any action!

Regards.

**Comment by ianbatesncc** [comment]
... I've suggested an implementation ... see #34  ... just for an idea.

**Comment by PHEgeorginaanderson** [comment]
Hi Ian.  Apologies this has taken so long to implement - I completely agree what you suggest is far more appropriate.  This is now resolved in v1.3.1.  The LE function now returns two new fields pops_used and dths_used and no longer returns the original pop and death columns passed as arguments to the function.
